### PR TITLE
[Task#45088] User - Create loaned books

### PR DESCRIPTION
**Pull request by TommyHoang2911**
## Related Tickets:
- [#45088](https://edu-redmine.sun-asterisk.vn/issues/45088)

## WHAT this PR does?

**Design logic back-end in function create loaned book for **User**.**

> 
> 1. Check valid user (role : member)
> 2. Check quantity loaned (maximum 1 user - 3 books)
> 3. Load data in Cart through the session.

## Attachments:

Screenshot from 2022-01-17 16-58-25

### CODE REVIEW

**Comment by TommyHoang2911** [comment]
ready

File: app/controllers/loaned_books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+class LoanedBooksController < ApplicationController
+  include SessionsHelper, CartsHelper
+
+  before_action :check_valid_user, :load_cart, :check_quantity_add, :assign_new_loan, :create_loan_detail, only: :create
```
Comment (vanvtt-0952) [inline app/controllers/loaned_books_controller.rb]: chưa thấy check việc user phải login chú nhỉ

Comment (TommyHoang2911) [reply app/controllers/loaned_books_controller.rb:4]:
```
def check_valid_user
    return if User.is_valid_member? current_user

    flash.now[:danger] = t "error.user_invalid"
    redirect_to home_path
  end
```
Chỗ current_user là method bên SessionHelper trả về current_user ấy chị. Em làm vậy ổn không chị hè ?

**Comment by TommyHoang2911** [comment]
ready

**Pull request merged**